### PR TITLE
Make Logger to log object and accept variadic params

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
While converting existing console.log to Logger's methods,
two changes were made.

1. Logging object type in Json format
2. Accept variadic params

This PR also has some usage cases of 1 and 2.

For https://github.com/Samsung/ONE-vscode/issues/662

ONE-vscode-DCO-1.0-Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

File: src/Backend/Backend.ts [hunk 1]
```diff
@@ -39,7 +41,7 @@ function backendRegistrationApi() {
       if (compiler) {
         gToolchainEnvMap[backend.name()] = new ToolchainEnv(compiler);
       }
-      console.log(`Backend ${backendName} was registered into ONE-vscode.`);
+      Logger.info(logTag, 'Backend', backendName, 'was registered into ONE-vscode.');
```
Comment (hyunsik-yoon) [inline src/Backend/Backend.ts:44]: image

File: src/Project/JobRunner.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export class JobRunner extends EventEmitter {
       tool = 'onecc';
     }
 
-    console.log('Run tool: ', tool, ' args: ', toolArgs, ' cwd: ', path, ' root: ', job.root);
+    Logger.info(this.tag, 'Run tool:', tool, 'args:', toolArgs, 'cwd:', path, 'root: ', job.root);
```
Comment (hyunsik-yoon) [inline src/Project/JobRunner.ts:69]: image

File: src/Project/BuilderCfgFile.ts [hunk 3]
```diff
@@ -150,7 +150,7 @@ export class BuilderCfgFile extends EventEmitter implements helpers.FileSelector
     let inputModel = path.basename(importTFlite.inputPath);
     importTFlite.name = 'ImportTFlite ' + inputModel;
 
-    console.log('importTFlite = ', importTFlite);
+    Logger.debug(this.tag, 'importTFlite = ', importTFlite);
```
Comment (hyunsik-yoon) [inline src/Project/BuilderCfgFile.ts:153]: image

**Review by yunjayh — APPROVED** [review]
LGTM Thank you!

File: src/Utils/Logger.ts [hunk 4]
```diff
@@ -16,6 +16,8 @@
 
 import * as vscode from 'vscode';
 
+type MsgList = (number|boolean|string|object)[];
```
Comment (YongseopKim) [inline src/Utils/Logger.ts:19]: Is it meaning that `MsgList` is a list for number, boolean, string and object?

Comment (YongseopKim) [reply src/Utils/Logger.ts:19]: If object is a type of the `MsgList`, it's meaning that the list is all of types.

File: src/Project/BuilderCfgFile.ts (hunk 3 above)
Comment (hyunsik-yoon) [reply src/Project/BuilderCfgFile.ts:153]: importTflite is object type.

File: src/Backend/Backend.ts (hunk 1 above)
Comment (hyunsik-yoon) [reply src/Backend/Backend.ts:44]: type of `backendName` is string.

File: src/Project/JobRunner.ts (hunk 2 above)
Comment (hyunsik-yoon) [reply src/Project/JobRunner.ts]: - `args`: array type
- `job.root`: boolean type
- `tool`, `path`: string type

File: src/Utils/Logger.ts (hunk 4 above)
Comment (hyunsik-yoon) [reply src/Utils/Logger.ts:19]: > Is it meaning that MsgList is a list for number, boolean, string and object?
> If object is a type of the MsgList, it's meaning that the list is all of types.

Yes. correct. :-)
I added more comment on the above samples: 
- https://github.com/Samsung/ONE-vscode/pull/710#discussion_r881416019 
- https://github.com/Samsung/ONE-vscode/pull/710#discussion_r881414291

**Review by jyoungyun — APPROVED** [review]
LGTM

File: src/Backend/Backend.ts (hunk 1 above)
Comment (hyunsik-yoon) [reply src/Backend/Backend.ts:44]: Note: Previous code actually does not need to be changed to this code.... but I changed just to make a sample where `string` can be passed as one of params.
Previous code (one single msg string) with `Logger.info` like
`Looger.info(tag, `Backend ${backendName} was registered into ONE-vscode.`);` also works.

**Review by YongseopKim — APPROVED** [review]
LGTM

**Pull request merged**